sansa-rdf-spark/src/main/scala/net/sansa_stack/rdf/spark/model/hdt/TripleOps.scala

```diff
@@ -0,0 +1,194 @@
+package net.sansa_stack.rdf.spark.model.hdt
+
+import net.sansa_stack.rdf.spark.io._
+import org.apache.jena.graph
+import org.apache.jena.riot.Lang
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Row, SaveMode, SparkSession}
+import org.apache.spark.sql.types.{LongType, StringType, StructField, StructType}
+
+
+class TripleOps {
+  var tripleFactTable : DataFrame = _
+  var objectDF : DataFrame = _
+  var predicateDF : DataFrame = _
+  var subjectDF : DataFrame = _
+  private val spark : SparkSession = SparkSession.builder().getOrCreate()
+
+  /**
+    * Function returns the Schema of Indexed Triple Fact table.
+    * @return StructType
+    */
+  def tripleSchema : StructType = {
+    StructType(
+      Seq( StructField(name = TripleOps.SUBJECT_COL, dataType = StringType, nullable = false),
+        StructField(name = TripleOps.OBJECT_COL, dataType = StringType, nullable = false),
+        StructField(name = TripleOps.PREDICATE_COL, dataType = StringType, nullable = false)))
+  }
+
+  /**
+    * Function returns the Schema of Dictionary Dataframe.
+    * @return Schema of Dictionary
+     */
+  def dictionarySchema : StructType = {
+    StructType(
+      Seq( StructField(name = TripleOps.NAME_COL, dataType = StringType, nullable = false),
+        StructField(name = TripleOps.INDEX_COL, dataType = LongType, nullable = false)))
+  }
+
+    /**
+    * Function converts RDD[graph.Triple] to DataFrame [Subject,Object,Predicate] by extracting SOP  value from each record
+    * @param triple: Input raw RDD[graph.Triple]
+    * @return Returns DataFrame [Subject,Object,Predicate]
+      */
+  def convertRDDGraphToDF(triple : RDD[graph.Triple]) : DataFrame = {
+    spark.createDataFrame(triple.map(t => Row(t.getSubject.toString, t. getObject.toString(), t.getPredicate.toString())), tripleSchema)
+  }
+
+
+   /**
+    * Registered input Subject, Object and Predicate Dataframe as Spark in-memory Views
+    * @param subjectDictDF Dictionary Dataframe of Subject
+    * @param objectDictDF Dictionary Dataframe of Object
+    * @param predicateDict Dictionary Dataframe of Predicate
+    */
+  def registerDictionariesAsView(subjectDictDF : DataFrame, objectDictDF : DataFrame, predicateDict : DataFrame) : Unit = {
+    subjectDictDF.createOrReplaceTempView(TripleOps.SUBJECT_TABLE)
+    objectDictDF.createOrReplaceTempView(TripleOps.OBJECT_TABLE)
+    predicateDict.createOrReplaceTempView(TripleOps.PREDICATE_TABLE)
+  }
+
+
+  /**
+    * Return Dataframe of Index + Subject by retrieving the unique subjects from RDD[Triple] and zip it with undex
+    * @param triples RDD[Triple] conversion of input file
+    * @return DataFrame Subject dictionary of [index,subject]
+    */
+  def getDistinctSubjectDictDF(triples : RDD[graph.Triple]) : DataFrame = {
+    spark.createDataFrame(triples.map(_.getSubject.toString()).distinct().zipWithIndex().map(t => Row(t._1, t._2)), dictionarySchema).cache()
```

Comment:
Do we need to persist/cache them in advance? I think would be good that we remove the caching as a default approach and leave it to the user how they see it relevant for their use case.
So, remove the `.cache()` the method from everywhere where it is added and which can be provided by the user.